client/packages/requisitions/src/ResponseRequisition/ListView/ProgramRequisitionOptions.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Autocomplete,
+  AutocompleteProps,
+  ButtonWithIcon,
+  Grid,
+  PlusCircleIcon,
+  Typography,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { getNameOptionRenderer } from '@openmsupply-client/system';
+
+import { CustomerProgramSettingsFragment } from '../api';
+import { NewRequisitionType } from '../../types';
+
+export interface NewProgramRequisition {
+  type: NewRequisitionType.Program;
+  programOrderTypeId: string;
+  otherPartyId: string;
+  periodId: string;
+}
+
+type Common<T> = Pick<
+  AutocompleteProps<T>,
+  'options' | 'value' | 'disabled' | 'renderOption' | 'getOptionDisabled'
+> & {
+  label: string;
+  set: (value: T | null) => void;
+  labelNoOptions?: string;
+};
+
+const useProgramRequisitionOptions = (
+  programSettings: CustomerProgramSettingsFragment[]
+) => {
+  const t = useTranslation();
+  type ProgramSetting = CustomerProgramSettingsFragment;
+  type CustomerAndOrderTypes =
+    CustomerProgramSettingsFragment['customerAndOrderTypes'][number];
+  type OrderType = CustomerAndOrderTypes['orderTypes'][number];
+  type Customer = CustomerAndOrderTypes['customer'];
+  type Period = OrderType['availablePeriods'][number];
+
+  const [program, setProgram] = useState<ProgramSetting | null>(null);
+  const [orderType, setOrderType] = useState<OrderType | null>(null);
+  const [customer, setCustomer] = useState<Customer | null>(null);
+  const [period, setPeriod] = useState<Period | null>(null);
+
+  useEffect(() => {
+    setOrderType(null);
+    setCustomer(null);
+  }, [program]);
+
+  useEffect(() => {
+    setPeriod(null);
+  }, [orderType]);
+
+  const allOptions: {
+    programs: Common<ProgramSetting>;
+    orderTypes: Common<OrderType>;
+    customers: Common<Customer>;
+    periods: Common<Period>;
+  } = {
+    programs: {
+      options: programSettings,
+      value: program,
+      set: setProgram,
+      label: t('label.program'),
+      disabled: false,
+    },
+    orderTypes: {
+      options: program?.customerAndOrderTypes.flatMap(c => c.orderTypes) || [],
+      value: orderType,
+      set: setOrderType,
+      disabled: program === null,
+      labelNoOptions: t('messages.not-configured'),
+      label: t('label.order-type'),
+    },
+    customers: {
```

Comment:
[5893d09](https://github.com/msupply-foundation/open-msupply/pull/5220/commits/5893d0917c96c59175842f304fbf8d87adb46be6) big enough?

Reply:
Using windowDimensions was causing it to re-render to original state... maybe because it's being used in ListSearch as well? Tried to use it asRef to see if that would retain the state.... have just used a hardcoded height 🤔  seems fine on tablet?